shim/handler.go

```diff
@@ -201,8 +195,7 @@ func (h *Handler) handleInit(msg *peer.ChaincodeMessage) (*peer.ChaincodeMessage
 		return nil, fmt.Errorf("failed to marshal response: %s", err)
 	}
 
-	err = h.sendBatch(msg.ChannelId, msg.Txid)
-	if err != nil {
+	if err := stub.FinishWriteBatch(); err != nil {
```

Comment:
you can only leave equals here

```
if err = stub.FinishWriteBatch(); err != nil {
```

Reply:
I disagree.

You're just creating an extra variable here, which you return immediately. 
Show me an example of how a new variable could be better here?

Reply:
If you are set on avoiding block scoped variables where possible, let's have a package global err value, right?

Reply:
I doubt the compiler will leave this code untouched

Reply:
Off the top of my head, an example of where the block scoped variable is better is that, if the code above is changed and the err remove, the code in this if block doesn't break. It is less fragile.

Reply:
Thank you.
For me the most important thing in go code is readability, simplicity and understandability of the code.
And when I see a shaded variable appearing, I stumble at it, and such code becomes less readable for me. I am always tormented by it. And yes, I like to save memory for small things.

But I won't insist on fixing it. Let's assume that there are my preferences and yours. And there are no general rules defined for this.